### PR TITLE
Handle incompatible proto change in frontend

### PR DESCRIPTION
**Pull request by dnr**
## What changed?
Handle proto compatibility mistake by checking unknown fields.

## Why?
Compatibility during deployments.

## How did you test it?
manually

### CODE REVIEW

File: service/frontend/workflow_handler.go [hunk 1]
```diff
@@ -2720,7 +2722,48 @@ func (wh *WorkflowHandler) DescribeTaskQueue(ctx context.Context, request *workf
 		return nil, err
 	}
 
-	return matchingResponse.DescResponse, nil
+	resp := matchingResponse.DescResponse
+	// Manually parse unknown fields to handle proto incompatibility.
+	// TODO: remove this after 1.24.0-m3
+	if resp == nil {
+		resp = &workflowservice.DescribeTaskQueueResponse{}
+		unknown := []byte(resp.ProtoReflect().GetUnknown())
```
Comment (ShahabT) [inline service/frontend/workflow_handler.go:2730]:
```suggestion
		unknown := []byte(matchingResponse.ProtoReflect().GetUnknown())
```

File: service/frontend/workflow_handler.go [hunk 2]
```diff
@@ -2720,7 +2722,50 @@ func (wh *WorkflowHandler) DescribeTaskQueue(ctx context.Context, request *workf
 		return nil, err
 	}
 
-	return matchingResponse.DescResponse, nil
+	resp := matchingResponse.DescResponse
```
Comment (tdeebswihart) [inline service/frontend/workflow_handler.go:2725]: Do we have test coverage for this logic?

Comment (dnr) [reply service/frontend/workflow_handler.go:2725]: I tested the relevant combinations manually. Do we need unit tests too? It's temporary code just to unstick this deployment

**Review by ShahabT — APPROVED** [review]
The reasoning is sound and code lgtm

File: service/frontend/workflow_handler.go (hunk 2 above)
Comment (ShahabT) [reply service/frontend/workflow_handler.go:2725]: Also not sure if unit test adds much value for this temporary code. We can do even more manual test in Cloud once the build is created.

**Pull request merged**